crates/mun_mdbook_test_plugin/src/main.rs

```diff
@@ -0,0 +1,155 @@
+use mdbook::renderer::RenderContext;
+use mdbook::BookItem;
+
+use std::io;
+
+use regex::Regex;
+
+use mun_test::{CompileAndRunTestDriver, CompileTestDriver};
+
+use mun_runtime::{invoke_fn, RuntimeBuilder};
+
+use std::panic;
+
+enum TestingMethod {
+    Skip,
+    CompileFail,
+    CompileAndRunFail,
+    Compile,
+    CompileAndRun,
+}
+
+impl Default for TestingMethod {
+    fn default() -> Self {
+        TestingMethod::Compile
+    }
+}
+
+fn get_testing_method(code: &str) -> TestingMethod {
+    let mut testing_method = TestingMethod::default();
+
+    let function_name_regex = Regex::new(r###"fn\("((?s:.)+)"\)"###).unwrap();
+
+    for line in code.lines() {
+        if let Some(first_char) = line.chars().next() {
+            if first_char != ',' {
+                return testing_method;
+            }
+        } else {
+            return testing_method;
+        }
+        if line == ",compile_fail" {
+            match testing_method {
+                TestingMethod::Compile => testing_method = TestingMethod::CompileFail,
+                TestingMethod::CompileAndRun => testing_method = TestingMethod::CompileAndRunFail,
+                _ => (),
+            }
+        }
+        if line == ",run" {
+            match testing_method {
+                TestingMethod::Compile => testing_method = TestingMethod::CompileAndRun,
+                TestingMethod::CompileFail => testing_method = TestingMethod::CompileAndRunFail,
+                _ => (),
+            }
+        }
```

Comment:
```suggestion
        if line == ",no_run" {
            testing_method = TestingMethod::Compile;
        }
```

Reply:
I think it's easier to understand for people if we stick to the same standards as mdbook: `no_run`

source: https://rust-lang.github.io/mdBook/format/mdbook.html#including-portions-of-a-file